src/net_processing.cpp

```diff
@@ -3347,6 +3347,14 @@ bool ProcessNetBlock(const CChainParams& chainparams, const std::shared_ptr<cons
     {
         LOCK(cs_main);
 
+        // Check that the coinstake transaction exist in the received block
+        if(pblock->IsProofOfStake() && !(pblock->vtx.size() > 1 && pblock->vtx[1]->IsCoinStake()))
```

Comment:
I think this is good to be paranoid about security, but was there a specific reason this was added?